corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js

```diff
@@ -22,7 +23,35 @@ hqDefine('reports/v2/js/datagrid/data_models', [
         self.isLoadingError = ko.observable(false);
 
         self.rows = ko.observableArray([]);
-        self.length = 50; // todo change number dynamically?
+
+        self.limit = ko.observable(undefined);
+        self.hasLimitBeenModified = ko.observable(false);
+
+        self.currentPage = ko.observable(undefined);
+        self.totalRecords = ko.observable(1);
+
+        self.limit.subscribe(function () {
+            // needed to stay in sync with pagination widget
+            // to prevent unnecessary reloads of data
+            self.hasLimitBeenModified(true);
+        });
+
+        self.goToPage = function (page) {
+            if (page !== self.currentPage() || self.hasLimitBeenModified()) {
+                self.currentPage(page);
+                self.hasLimitBeenModified(false);
+                self.loadRecords();
+            }
+        };
+
+        self._pushState = function () {
+            var stateInfo = {
+                    page: self.currentPage(),
+                    limit: self.limit(),
+                },
+                url = '?p=' + self.currentPage() + "&l=" + self.limit();
```

Comment:
no...with this setup other parameters will be either part of the base url (like a config id) or passed as a`POST` parameter

Reply:
Doesn't that mean you will never be able to send a link to a specifically filtered report? For the original we went with storing things in URL params so that people can bookmark and send links around: https://staging.commcarehq.org/a/polar-bears-are-cool/reports/case_list_explorer/?search_xpath=(favorite_number%20%3E%3D%203%20or%20(%40case_type%20%3D%20%27adult%27%20and%20blood_group%20%3D%20%27AB%2B%27))%20and%20(name%20!%3D%20%27Aaron%20Baldwin%27)&explorer_columns=[%22%40case_type%22%2C%22case_name%22%2C%22last_modified%22%2C%22favorite_number%22%2C%22blood_group%22]&case_list_filter=project_data&case_type=&is_open=